tests/test_generate_marketplace.py

```
@@ -91,25 +110,215 @@ def test_generate_marketplace_builds_plugin_groups(tmp_path: Path) -> None:
+
+    mod = _load_generate_marketplace_module()
+    _write_skill(repo_root=tmp_path, skill_id="alpha")
+    _write_skill(repo_root=tmp_path, skill_id="beta")
+    _write_bundles(repo_root=tmp_path, body=_core_bundles_yaml())
+    _write_version(repo_root=tmp_path, version="4.5.6")
+    tmp_path.joinpath("pyproject.toml").write_text(
+        '[project]\nname = "fake"\nversion = "9.9.9"\n',
+        encoding="utf-8",
     )
 
+    with pytest.raises(ValueError, match="does not match pyproject.toml"):
```

Ruff (0.16.2)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)